Feature/#155113216

ejaros opened the pull request.

Commit:
wip refactor to drench_sdk

Commit:
remove statemachine dir

Commit:
wip add placeholder taxonomy and resource classes; exporting json

Commit:
alwas append sns flows

Commit:
use resource.py from drench api, contitionally attach sns on topic arn presence

Commit:
un-nest success and fail states

Commit:
rename resource -> resources to avoid colisions, use resources.get_arn throughout

Commit:
add basic taxonomy checking

Commit:
remove topic_arn from workflows

Commit:
correct imports for packaging

Commit:
add initial integration test

Commit:
add todo to README

Commit:
wip lint flows"

Commit:
have flow inherit from state

Commit:
remove redundant ResultPath arguments, restore tests

Commit:
add taxonomy error check

Commit:
wip - resource.py hacks creating a runnable sfn

Commit:
add resource_type to arns

Commit:
rename flow -> transform and add athena query transform and lambdas

Commit:
rename workflow.addFlow -> workflow.addTransForm

Commit:
wip add query lambdas

Commit:
correct boto call in check_query

Commit:
corrected test target

Commit:
remove old examples, prune setup.py

Commit:
remove old examples

coreycollins commented on drench_sdk/resources.py:
```diff
@@ -0,0 +1,21 @@
+        #        rather than
+        #   resourcetype:resource
+        # see https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+        return f"arn:{partition}:{service}:{region}:{account_id}:{resource_type}:{resource}"
```
Instead of requiring a resource and a resource_type, let's just expect the method caller to know how to specific the resource. That way we can just combine the arguments.

coreycollins commented on drench_sdk/states.py:
```diff
@@ -1,18 +1,17 @@
-
-    def __init__(self, Type, InputPath=None, OutputPath=None, Next=None, End=False):
+    def __init__(self, Type, InputPath=None, ResultPath=None, OutputPath=None, Next=None, End=False):
         self.Type = Type
```
Should the State class contain a `Start` class argument instead of the subclass Transform?

coreycollins commented on drench_sdk/transforms.py:
```diff
@@ -0,0 +1,253 @@
+        self.TopicArn = TopicArn
+        self.Subject = Subject
+        self.Message = Message
+        self.can_fail = False
```
Can you walk me through what `can_fail` does?

coreycollins commented on drench_sdk/transforms.py:
```diff
@@ -0,0 +1,253 @@
+        self.QueryString = QueryString
+        self.QueryExecutionContext = QueryExecutionContext
+        self.ResultConfiguration = ResultConfiguration
+        self.on_fail = None
```
Why do we set this to None? It looks like it can never be set to any value? It would be better if the parent Transform class at least set this from a default None __init__ argument .

Commit:
implement changes from code review
 - remove can_fail from SNSTransform
 - add on_fail to Transform constructor
 - remove resource_type from get_arn
 - move Start poperty from Transform to State
